qualtran/resource_counting/_call_graph.py

```diff
@@ -69,15 +70,19 @@ def build_cbloq_call_graph(cbloq: CompositeBloq) -> Set[BloqCountT]:
 
 
 def _generalize_callees(
-    raw_callee_counts: Set[BloqCountT], generalizer: GeneralizerT
+    raw_callee_counts: Union[BloqCountDictT, Set[BloqCountT]], generalizer: GeneralizerT
 ) -> List[BloqCountT]:
     """Apply `generalizer` to the results of `bloq.build_call_graph`.
 
     This calls `generalizer` on each of the callees returned from that function,
     and filters out cases where `generalizer` returns `None`.
     """
     callee_counts: Dict[Bloq, Union[int, sympy.Expr]] = defaultdict(lambda: 0)
-    for callee, n in raw_callee_counts:
+    if isinstance(raw_callee_counts, dict):
+        raw_callee_iterator: Iterable[BloqCountT] = raw_callee_counts.items()
+    else:
+        raw_callee_iterator = raw_callee_counts
```

Comment:
this would be a good place to emit a deprecation warning if a set is encountered

Reply:
Do you think we should wait on this until migration of most of the bloqs is complete?
I can add this, but it might be a bit noisy for a while.

Reply:
we can wait